gocourse18/mysql/init.go

```diff
@@ -0,0 +1,139 @@
+package mysql
+
+import (
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/jmoiron/sqlx"
+	"github.com/vikibell/prjctr-golang-beginning/gocourse18/domains/statistic"
+	"github.com/vikibell/prjctr-golang-beginning/gocourse18/service"
+	"gorm.io/driver/mysql"
+	"gorm.io/gorm"
+	"log/slog"
+)
+
+const DSN = "vika:123@tcp(127.0.0.1:3320)/db?parseTime=true"
+
+var usersSchema = `
+CREATE TABLE IF NOT EXISTS users
+(
+    id         bigint unsigned auto_increment primary key,
+    name       longtext         not null,
+    surname    longtext         not null,
+    email      longtext         not null,
+    age        bigint default 0 not null,
+    sex        longtext         not null,
+    city       longtext         not null,
+    taxi_count bigint default 0 null,
+    profession longtext         null
+);`
+
+var statisticsSchema = `
+CREATE TABLE IF NOT EXISTS statistics
+(   
+    id            bigint unsigned auto_increment primary key,
+    city          longtext         not null,
+    average_trips bigint default 0 null,
+    age_range     int    default 0 null
+);`
+
+type DB struct {
+	Connection *gorm.DB
+}
+
+func NewDB(dsn string) *DB {
+	gormdb, err := gorm.Open(mysql.Open(dsn), &gorm.Config{})
+	if err != nil {
+		panic("failed to connect database")
+	}
+
+	return &DB{
+		Connection: gormdb,
+	}
+}
+
+func init() {
+	populateDB()
+}
+
+func populateDB() {
+	db, err := sqlx.Connect("mysql", DSN)
+	if err != nil {
+		panic("Failed to connect database")
+	}
+
+	db.MustExec(usersSchema)
+	populateUsers(db)
+	db.MustExec(statisticsSchema)
+	populateStatistics(db)
+	db.Close()
+}
+
+func populateUsers(db *sqlx.DB) {
+	query := `TRUNCATE TABLE users`
+	_, trancErr := db.Exec(query)
+	if trancErr != nil {
+		slog.Error("Failed to truncate table", "error", trancErr)
+		return
+	}
+
+	users := service.GetUsersFromFile()
+
+	tx := db.MustBegin()
+	_, insertErr := tx.NamedExec(`
+		INSERT INTO users (name, surname, email, age, sex, city, taxi_count, profession) 
+		VALUES (:name, :surname, :email, :age, :sex, :city, :taxi_count, :profession)`,
+		users)
+	if insertErr != nil {
+		slog.Error("Failed to insert into table", "error", insertErr)
+	}
+
+	commitErr := tx.Commit()
+	if commitErr != nil {
+		slog.Error("Failed to commit changes", "error", commitErr)
+		return
+	}
+}
+
+func populateStatistics(db *sqlx.DB) {
```

Comment:
Оця функція може повертати помилку, а вже вище відбуватиметься логування.

```suggestion
func populateStatistics(db *sqlx.DB) error {
```